### PR TITLE
Rightclick to set and unset your respawn at a bonfire

### LINKED ISSUE #8: Rightclick to set and unset your respawn at a bonfire
Right now, rightclicking a bonfire sets the respawn and leftclicking unsets it. Make both actions rightclick.

### PR DESCRIPTION
**Pull request by pixelfac**
resolve #8

### CODE REVIEW

File: src/main/java/com/mineinabyss/bonfire/listeners/PlayerListener.kt [hunk 1]
```diff
@@ -46,16 +46,16 @@ object PlayerListener : Listener {
...
 
         if (clicked.type == Material.CAMPFIRE) {
             val respawnCampfire = clicked.state as Campfire
             val bonfire = respawnCampfire.bonfireData() ?: return
 
-            if (leftClicked) {
-                if (!player.removeBonfireSpawnLocation(bonfire.uuid)) {
-                    player.error("This is not your respawn point")
-                }
+            if (player.fallDistance > BonfireConfig.data.minFallDist){
```
Comment (Norazan) [inline src/main/java/com/mineinabyss/bonfire/listeners/PlayerListener.kt:57]: duplicate ccode with line 62

File: src/main/java/com/mineinabyss/bonfire/listeners/PlayerListener.kt [hunk 2]
```diff
@@ -65,14 +65,25 @@ object PlayerListener : Listener {
...
 
             transaction {
-                val playerCount = Players.select { bonfireUUID eq bonfire.uuid }.count()
-                if (playerCount >= BonfireConfig.data.maxPlayerCount) {
-                    return@transaction player.error("This bonfire is full!")
-                }else{
-                    player.setRespawnLocation(bonfire.uuid)
+                val playerBonfireUUID = Players
+                    .select { Players.playerUUID eq player.uniqueId }
+                    .firstOrNull()?.get(bonfireUUID) ?: return@transaction
```
Comment (Norazan) [inline src/main/java/com/mineinabyss/bonfire/listeners/PlayerListener.kt:70]: if a player has not set their respawn at a bonfire before, there will be no entry of them in the DB, meaning that `firstOrNull()` returns null. We want to set the player respawn in this case, but returning from the transaction doesn't do that. 

You can perhaps change this code to only get the first player row with `.firstOrNull()` without getting the bonfire UUID after, and then changing the if statements below to check `if (playerRow == null or playerRow.get(bonfireUUID) != bonfire.uuid) {set respawn} else {unset respawn}`

**Review by Norazan — CHANGES_REQUESTED** [review]
Looks good, just need to remove some duplicate code and account for null entries in the Players table.

File: src/main/java/com/mineinabyss/bonfire/listeners/PlayerListener.kt (hunk 1 above)
Comment (pixelfac) [reply src/main/java/com/mineinabyss/bonfire/listeners/PlayerListener.kt:57]: ez fix

File: src/main/java/com/mineinabyss/bonfire/listeners/PlayerListener.kt (hunk 2 above)
Comment (pixelfac) [reply src/main/java/com/mineinabyss/bonfire/listeners/PlayerListener.kt]: should be fixed

**Pull request merged**